WordPress/WordPressApi/WordPressComOAuthClient.swift

```diff
@@ -0,0 +1,189 @@
+
+    private let sessionManager: AFHTTPSessionManager
+    private let clientID: String
+    private let secret: String
+
+    /// Creates a WordPresComOAuthClient initialized with the clientID and secret constants defined in the ApiCredentials singleton
+    public class func client() -> WordPressComOAuthClient {
+        let client = WordPressComOAuthClient(clientID:ApiCredentials.client(), secret: ApiCredentials.secret())
+        return client
+    }
+
+    /**
+     Creates a WordPressComOAuthClient using the defined clientID and secret
+
+     - Parameters:
+       - clientID the app oauth clientID
+       - secret the app secret
+     */
+    public init(clientID: String, secret: String) {
+        self.clientID = clientID
+        self.secret = secret
+        let baseURL = NSURL(string:self.dynamicType.WordPressComOAuthBaseUrl)
+        let sessionConfiguration = NSURLSessionConfiguration.ephemeralSessionConfiguration()
+        sessionManager = AFHTTPSessionManager(baseURL:baseURL, sessionConfiguration:sessionConfiguration)
+        sessionManager.responseSerializer = AFJSONResponseSerializer()
+        sessionManager.requestSerializer.setValue("application/json", forHTTPHeaderField:"Accept")
+    }
+
+    /**
+     Authenticates on WordPress.com with Multifactor code.
+
+     - Parameters:
+       - username the account's username.
+       - password the account's password.
+       - multifactorCode Multifactor Authentication One-Time-Password. If not needed, can be nil
+       - success block to be called if authentication was successful. The OAuth2 token is passed as a parameter.
+       - failure block to be called if authentication failed. The error object is passed as a parameter.
+     */
+    public func authenticateWithUsername(username: String,
+                                  password: String,
+                                  multifactorCode: String?,
+                                  success:(authToken: String) -> (),
+                                  failure:(error: NSError) -> () )
+    {
+        var parameters: [String:AnyObject] = [
+            "username": username,
+            "password": password,
+            "grant_type": "password",
+            "client_id": ApiCredentials.client(),
+            "client_secret": ApiCredentials.secret(),
+            "wpcom_supports_2fa": true
+        ]
+
+        if let multifactorCode = multifactorCode where !multifactorCode.isEmpty() {
+            parameters["wpcom_otp"] = multifactorCode;
+        }
+
+        sessionManager.POST("token", parameters: parameters, success: { (task, responseObject) in
+            DDLogSwift.logVerbose("Received OAuth2 response: \(self.cleanedUpResponseForLogging(responseObject))")
+            guard let responseDictionary = responseObject as? [String:AnyObject],
+                let authToken = responseDictionary["access_token"] as? String else {
+                    success(authToken: "")
```

Comment:
The previous implementation would return `nil` if it couldn't parse the token properly. We should do the same (at least for this PR)